specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1409 @@
+Non-packaged applications need to use the 'boostrapper API' in ProjectReunion.Loader.dll to get access to
+ProjectReunion's Framework package. See [3.4. Runtime 'Pinning' aka Prevent Update While In-Use] for more details.
+
+The 'bootstrapper API' provides an initialization function that performs the following actions:
+
+1. Find the per-application 'helper' Main package and create a process with its identity
+2. Find and load ProjectReunion.dll from the ProjectReunion Framework package
+
+### 5.4.1. Boostrapper - Find and Load/Run the per-application 'helper' Main package
+
+The per-application 'helper' Main package defines a [COM OutOfProcess
+server](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-com-comserver)
+and `<PackageDependency Name="ProjectReunionFramework"...>`.
+
+The bootstrapper API calls CoCreateInstance() to create a process with this package's identity.
+The Dynamic Dependencies API can be used by non-packaged processes as long as that process runs.
+The bootstrapper API holds a reference to the returned object until a 'bootstrapper shutdown' API
+is called or process termination.
+
+### 5.4.2. Boostrapper - Find and Load ProjectReunion.dll
+
+The bootstrapper API finds the ProjectReunion Framework package and calls LoadLibrary("ProjectReunion.dll")
+to load and initialize the Dynamic Dependencies API.
+
+## 5.5. Dynamic Dependencies vis a vis Static Dependencies
+
+Dynamic dependencies' package dependency is equivalent to an MSIX package's `<PackageDependency>` in
+`appxmanifest.xml`. This defines criteria to be resolved to a package for use at runtime in a
+process' package graph.
+
+Packages dynamically added to a package graph can be removed. Packages in the package graph at
+process creation (aka static package graph) cannot be removed; the static package graph is a part of
+a process' DNA, from birth to death.
+
+## 5.6. Known Limitations in v1
+
+The initial implementation does not fully implement this spec.
+
+### 5.6.1. [uap6:LoaderSearchPathOverride](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-uap6-loadersearchpathoverride) not supported
+
+See [3.2.1. Known Issue: DLL Search Order ignores uap6:LoaderSearchPathOverride](#321-known-issue-dll-search-order-ignores-uap6loadersearchpathoverride).
+
+### 5.6.2. [uap6:AllowExecution](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-uap6-allowexecution) not supported
+
+See [3.2.2. Known Issue: DLL Search Order ignores uap6:AllowExecution](#322-known-issue-dll-search-order-ignores-uap6allowexecution).
+
+# 6. API Details
+
+**TODO** Retitle this section 'API' since we don't have 'API Notes'?
+
+## 6.1. Win32 API - MsixDynamicDependency.hpp
+
+All Win32 APIs are prefixed with Mdd/MDD for MSIX Dynamic Dependencies.
+
+```c++
+enum class MddPinPackageDependency : uint32_t
```

Comment:
Compiler won't accept same named enum+function e.g. `enum foo{...}` and `void foo();` or at least not `extern "C" void foo();`

Rename to MddPinPackageDependencyOptions or MddPinPackageDependencyFlags